src/lockscreen.js

```diff
@@ -156,13 +137,15 @@ var Lockscreen = GObject.registerClass({
         if (!assetsPath)
             assetsPath = defaultPath;
 
-        if (videoPath)
-            this._openURI = `file://${videoPath}/open.webm`;
-        else
-            this._openURI = null;
+        return [assetsPath, videoPath, assetsHasKey];
+    }
 
-        this._playbin.hasKey = this._key && this._manager.hasKey(this._key);
+    _updateBackground() {
+        const [assetsPath, videoPath, assetsHasKey] = this.getAssetsPath();
 
+        this._openURI = `file://${videoPath}/open.webm`;
+
+        this._playbin.hasKey = this._key && this._manager.hasKey(this._key);
```

Comment:
This looks a bit unclear, unless the `_manager.hasKey()` method has a side effect, it looks like it's not doing anything?

Reply:
Yes this was already there. I think it's wrong. I will change it.